File: rust/src/problems/problem_217.rs
// https://leetcode.com/problems/contains-duplicate/description/
use std::collections::HashMap;

#[allow(dead_code)]
pub fn contains_duplicate(nums: Vec<i32>) -> bool {
    let mut hm: HashMap<i32, bool> = HashMap::new();

    for n in nums {
        if hm.contains_key(&n) {
            return true;
        }
        hm.insert(n, true);
    }

    false
}

#[cfg(test)]
mod tests {
    use crate::problems::problem_217::contains_duplicate;

    #[test]
    fn p217() {
        assert_eq!(contains_duplicate(vec![1, 2, 3, 1]), true);
        assert_eq!(contains_duplicate(vec![1, 2, 3, 4]), false);
        assert_eq!(contains_duplicate(vec![1, 1, 1, 3, 3, 4, 3, 2, 4, 2]), true);
    }
}

File: Kotlin/app/src/main/kotlin/problems/Problem217.kt
import Utils.*

// https://leetcode.com/problems/contains-duplicate/description/

fun containsDuplicate(nums: IntArray): Boolean {
    val reps = mutableMapOf<Int, Boolean>()

    for (n in nums) {
        if (reps.containsKey(n)) {
            return true
        }
        reps[n] = true
    }

    return false
}

fun test217() {
    println("217) Contains Duplicate:")
    val ts =
        mapOf(
            intArrayOf(1, 2, 3, 1) to true,
            intArrayOf(1, 2, 3, 4) to false,
            intArrayOf(1, 1, 1, 3, 3, 4, 3, 2, 4, 2) to true,
        )

    for ((input, expected) in ts) {
        val res = containsDuplicate(input) == expected
        println("For ${input.contentToString()}: ${colorResult(res)}")
    }
}
